src/pretix/presale/forms/checkout.py

```diff
@@ -299,6 +300,27 @@ def __init__(self, *args, **kwargs):
                     initial=initial.file if initial else None,
                     widget=UploadedFileWidget(position=pos, event=event, answer=initial)
                 )
+            elif q.type == Question.TYPE_DATE:
+                field = forms.DateField(
+                    label=q.question, required=q.required,
+                    help_text=q.help_text,
+                    initial=initial.answer if initial else None,
+                    widget=DatePickerWidget(),
+                )
+            elif q.type == Question.TYPE_TIME:
+                field = forms.TimeField(
+                    label=q.question, required=q.required,
+                    help_text=q.help_text,
+                    initial=initial.answer if initial else None,
```

Comment:
``initial=initial.answer if initial else None`` probably needs to be ``initial=dateutil.parser.parse(initial.answer).astimezone(tz).time() if initial and initial.answer else None``